### PR TITLE
Adding support for RESQML 2.2 SeismicWellboreFrameRepresentation

### PR DESCRIPTION
**Pull request by mpoudret**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Adding support for RESQML 2.2 SeismicWellboreFrameRepresentation:
- adding RESQML 2.2 SeismicWellboreFrameRepresentation support in fesapi (WellboreFrameRepresentation is now in both resqml2/resqml2_0_1/resqml2_2 classes)
- updating/adding corresponding unit tests
- updating/adding SWIG wrappers
- updating C# and Java examples

Does this close any currently open issues?
------------------------------------------
No.

Any relevant logs, error output, etc?
-------------------------------------
No.

Any other comments?
-------------------
No.

Where has this been tested?
---------------------------
**Operating System:** Windows 10 64bits

**Platform:** Visual Studio 2019/Eclipse 2019-06

### CODE REVIEW

File: src/tools/Misc.cpp [hunk 1]
```diff
@@ -33,3 +34,17 @@ string misc::getPartNameFromReference(gsoap_eml2_2::eml22__DataObjectReference *
 	return reference->ContentType.substr(reference->ContentType.rfind('=') + 1) + "_" + reference->Uuid + ".xml";
 }
 
+gsoap_resqml2_0_1::eml20__DataObjectReference* misc::eml22ToEml20Reference(gsoap_eml2_2::eml22__DataObjectReference* reference, soap* soapContext)
+{
+	gsoap_resqml2_0_1::eml20__DataObjectReference* result = gsoap_resqml2_0_1::soap_new_eml20__DataObjectReference(soapContext);
+	result->UUID = reference->Uuid;
+	result->Title = reference->Title;
+	result->ContentType = reference->ContentType;
+	if (reference->ObjectVersion != nullptr) {
```
Comment (philippeVerney) [inline src/tools/Misc.cpp]: could also check if it is not empty neither.
An emtpy version does not mean anything and, imo, is like a null version.

File: src/common/DataObjectRepository.h [hunk 2]
```diff
@@ -1,6 +1,6 @@
 /*-----------------------------------------------------------------------
 Licensed to the Apache Software Foundation (ASF) under one
-or more contributor license agreements.  See the NOTICE file
+or more contributor license agreements.  See the NOTICE createWellboreMarkerFrameRepresentation
```
[the remark this answers is not in the document]
Comment (mpoudret) [reply src/common/DataObjectRepository.h]: Done.

File: src/common/DataObjectRepository.h [hunk 3]
```diff
@@ -273,7 +275,9 @@ namespace COMMON_NS
 		template <class valueType>
 		std::vector<valueType *> getSourceObjects(COMMON_NS::AbstractObject const * dataObj) const
 		{
-			return getObjsFilteredOnDatatype<valueType>(getSourceObjects(dataObj));
+			const std::vector < COMMON_NS::AbstractObject*> sourceObjects = getSourceObjects(dataObj);
```
[the remark this answers is not in the document]
Comment (mpoudret) [reply src/common/DataObjectRepository.h]: Done.

File: src/resqml2_2/SeismicWellboreFrameRepresentation.cpp [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+	xmlTimeValues->Values->ExternalFileProxy[0]->StartIndex = 0;
+	xmlTimeValues->Values->ExternalFileProxy[0]->PathInExternalFile = "/RESQML/" + frame->uuid + "/timeValues";
+	xmlTimeValues->Values->ExternalFileProxy[0]->EpcExternalPartReference = proxy->newEml22Reference();
+
+	frame->NodeTimeValues = xmlTimeValues;
+
+	frame->NodeCount = timeValueCount;
+
+	// HDF
+	hsize_t dim[] = { timeValueCount };
```
[the remark this answers is not in the document]
Comment (mpoudret) [reply src/resqml2_2/SeismicWellboreFrameRepresentation.cpp]: Done.

File: src/resqml2_2/SeismicWellboreFrameRepresentation.cpp [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+			values[inc] = values[0] + (inc * constantArray->Value);
+	}
+	else {
+		throw logic_error("The array structure of time values is not supported?");
+	}
+}
+
+double SeismicWellboreFrameRepresentation::getSeismicReferenceDatum() const
+{
+	_resqml22__SeismicWellboreFrameRepresentation* frame = static_cast<_resqml22__SeismicWellboreFrameRepresentation*>(gsoapProxy2_2);
```
[the remark this answers is not in the document]
Comment (mpoudret) [reply src/resqml2_2/SeismicWellboreFrameRepresentation.cpp]: Done.

File: src/resqml2_2/SeismicWellboreFrameRepresentation.cpp [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+
+	return frame->SeismicReferenceDatum;
+}
+
+double SeismicWellboreFrameRepresentation::getWeatheringVelocity() const
+{
+	_resqml22__SeismicWellboreFrameRepresentation* frame = static_cast<_resqml22__SeismicWellboreFrameRepresentation*>(gsoapProxy2_2);
```
[the remark this answers is not in the document]
Comment (mpoudret) [reply src/resqml2_2/SeismicWellboreFrameRepresentation.cpp]: Done.

File: src/resqml2_2/SeismicWellboreFrameRepresentation.h [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+		DLL_IMPORT_OR_EXPORT void setTimeValues(double firstTimeValue, double incrementTimeValue, unsigned int timeValueCount);
+
+		/**
+		* Indicates either the times are regularly spaced or not (useful for optimization)
+		* Does not verify if the writer has used a generic array to store regular times.
+		*/
+		DLL_IMPORT_OR_EXPORT bool areTimeValuesRegularlySpaced() const;
+
+		/**
+		* Indicates the incremetn value between two times only if the times are regualrly spaced.
```
[the remark this answers is not in the document]
Comment (mpoudret) [reply src/resqml2_2/SeismicWellboreFrameRepresentation.h]: Done. Also line 79 of resqml2/WellboreFrameRepresentation.h.

File: src/tools/Misc.cpp (hunk 1 above)
Comment (mpoudret) [reply src/tools/Misc.cpp]: Done.

**Pull request merged**